bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,939 @@
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+		return nil
+
+	}); err != nil {
+		log.Errorf("Fetching list failed %s", err)
+		return RegisterProvisioningProfileResponseData{}, err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+	logDebugPretty(requestResponse)
+
+	return requestResponse.Data, nil
+}
+
+// UploadProvisioningProfile ...
+func (client *BitriseClient) UploadProvisioningProfile(uploadURL string, uploadFileName string, outputDirPath string, exportFileName string) error {
+	fmt.Println()
+	log.Infof("Upload %s to Bitrise...", exportFileName)
+
+	requestURL := uploadURL
+
+	files := map[string]string{
+		uploadFileName: (outputDirPath + "/" + exportFileName),
+	}
+
+	request, err := createUploadRequest("PUT", requestURL, map[string]string{}, files)
+	if err != nil {
+		return err
+	}
+
+	// Response struct
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		return nil
+
+	}); err != nil {
+		return err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+
+	return nil
+}
+
+// ConfirmProvisioningProfileUpload ...
+func (client *BitriseClient) ConfirmProvisioningProfileUpload(profileSlug string, provUploadName string) error {
+	fmt.Println()
+	log.Infof("Confirm - %s - upload to Bitrise...", provUploadName)
+
+	requestURL, err := urlutil.Join(baseURL, appsEndPoint, client.selectedAppSlug, provisioningProfilesEndPoint, profileSlug, "uploaded")
+	if err != nil {
+		return err
+	}
+
+	headers := map[string]string{
+		"Authorization": "token " + client.accessToken,
+	}
+
+	request, err := createRequest("POST", requestURL, headers, map[string]interface{}{})
+	if err != nil {
+		return err
+	}
+
+	// Response struct
+	responseStatusCode := -1
+	requestResponse := ConfirmProvProfileUploadResponse{}
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+
+		logDebugPretty(requestResponse)
```

Comment:
in previous requests this was at the end of the request not in the retries, is it ok here?